RFC-0003-arrow-flight-connector.md

```diff
@@ -71,6 +72,36 @@ Flight descriptor will be created from a command of byte array. The concrete imp
 
 ![Class diagram](arrow-flight-connector/Class-diagram.png)
 
+
+## Prestissimo Implementation
+A single Prestissimo connector will handle all concrete instances of the Java Arrow Flight connector. This is possible because the prestissimo connector does not have to deal with metadata fetch and discovery. Each split wraps a ticket and a list of locations, which is all[^1] that is required to fetch data from the Flight Service in an implementation-agnostic way.
+
+[^1]: Authentication is one exception which cannot be handled in a truly generic manner. A solution is proposed below.
+
+### Arrow Record Batch to Velox vectors
+Arrow arrays and Velox vectors binary compatible with the Arrow C Data Interface.
+- Velox provides bridge methods to convert Velox vectors to-and-from from the C Interface [(Link)](https://github.com/facebookincubator/velox/blob/main/velox/vector/arrow/Bridge.h)
+- The Arrow SDK provides bridge method to convert Arrow arrays to-and-from the C Interface [(Link)](https://arrow.apache.org/docs/cpp/api/c_abi.html)
+
+The Arrow Flight SDK returns data in the form of Arrow Arrays, which can be converted into Velox vectors by using the provided bridge methods, using the Arrow C Interface as an intermediate representation. In most cases, these are cheap operations that don't involve copying or translating the actual data buffers.
+
+### Plugin-style authentication
+The worker needs to authenticate with the Flight server to redeem the Flight Ticket. This cannot be handled in a truly generic manner since the Arrow Flight protocol allows a lot of freedom to implement custom authentication strategies. Each implementaion of a flight service may require custom code to handle authentication.
+
+We propose to solve this by adding plugin-style authenticators to the prestissimo connectors. A Flight catalog can optionally specify an authenticator to use:
+```properties
+connector.name=arrow-flight
+flight.auth.name=simple-user-pass
+# following are authenticator-specific properties
+flight.auth.user=me
+flight.auth.password=itsme
+```
+In the above case, the authenticator registered as `simple-user-pass` will be used to authenticate all Flight requests for that catalog. More authenticators can be created and registered using an interface similar to the interface for connector creation and registration in Velox.
+
+### Conenctor registration
+All concrete implementations of the presto connector will map to the same connector in prestissimo. `connector.name` values of `arrow-flight`, `arrow-flight-ibm`, `arrow-flight-sql` would all reference the same connector implementation.
```

Comment:
Is this connector name same as the one used in Presto connecotr.

Reply:
Yes

Reply:
So why refer to different connector names. We only have `ibm-arrow-flight` in Presto.

Reply:
If anyone creates another flight connector using the `presto-base-arrow-flight` module, they will need a prestissimo connector as well. They can reuse the same prestissimo connector for all variations of the Java Flight connector (except for authentication)

Reply:
Also, weren't we planning to create a Flight SQL connector as well?

Reply:
The RFC is only going to be for the base module, the template.

Reply:
Even if we do it that way, we still need to indicate to people how they can use their concrete Flight connectors with the same prestissimo connector

Reply:
As discussed, clarified slightly